packages/connections/src/utils/validation-errors.ts

```diff
@@ -0,0 +1,148 @@
+import { ConnectionInfo } from 'mongodb-data-service';
+import ConnectionString from 'mongodb-connection-string-url';
+import {
+  FormValidationError,
+  getConnectionString,
+  isSecure,
+} from './validation';
+
+function validateConnectionInfoErrors(
+  connectionInfo: ConnectionInfo
+): FormValidationError[] {
+  const connectionString = getConnectionString(connectionInfo);
+
+  return [
+    ...validateAuthMechanism(connectionString),
+    ...validateUsernameAndPassword(connectionString),
+    ...validateLocalhostAndTls(connectionString),
+    ...validateSSHTunnel(connectionInfo),
+    ...validateKerberosPrincipal(connectionString),
+    ...validateLDAP(connectionString),
+    ...validateX509(connectionString),
+  ];
+}
+
+function validateAuthMechanism(
+  connectionString: ConnectionString
+): FormValidationError[] {
+  const errors: FormValidationError[] = [];
+  return errors;
+}
+
+function validateX509(
+  connectionString: ConnectionString
+): FormValidationError[] {
+  const errors: FormValidationError[] = [];
+  if (connectionString.searchParams.get('authMechanism') === 'MONGODB-X509') {
```

Comment:
fyi, if https://github.com/mongodb-js/compass/pull/2651 is merged first it might be nice to update the core here to have similar changes :)